.github/workflows/failureNotifier.yml

```
@@ -0,0 +1,110 @@
+name: Notify on Workflow Failure
+
+on:
+  workflow_run:
+    workflows: ["Process new code merged to main"]
+    types:
+      - completed
+
+permissions:
+  issues: write
+
+jobs:
+  notify-failure:
+    runs-on: ubuntu-latest
+    if: ${{ github.event.workflow_run.conclusion == 'failure' }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Fetch Workflow Run Details
+        id: fetch-workflow-details
+        uses: actions/github-script@v7
+        with:
+          script: |
+            const runId = "${{ github.event.workflow_run.id }}";
+            const runData = await github.rest.actions.getWorkflowRun({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              run_id: runId
+            });
+            return runData.data;
+
+      - name: Fetch Workflow Run Jobs
+        id: fetch-workflow-jobs
+        uses: actions/github-script@v7
+        with:
+          script: |
+            const runId = "${{ github.event.workflow_run.id }}";
+            const jobsData = await github.rest.actions.listJobsForWorkflowRun({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              run_id: runId,
+            });
+            return jobsData.data;
+      
+      - name: Get merged pull request number
+        id: getMergedPullRequestNumber
+        uses: actions-ecosystem/action-get-merged-pull-request@59afe90821bb0b555082ce8ff1e36b03f91553d9
+        with:
+          github_token: ${{ github.token }}
+
+
+      - name: Process Each Failed Job
+        uses: actions/github-script@v7
+        with:
+          script: |
+            const prNumber = ${{ steps.getMergedPullRequestNumber.outputs.number }};
+            const runData = ${{fromJSON(steps.fetch-workflow-details.outputs.result)}};
+            const jobs = ${{steps.fetch-workflow-jobs.outputs.result}};
```

```suggestion
            const jobs = ${{ steps.fetch-workflow-jobs.outputs.result }};
```
Can you tidy up whitespace like this elsewhere too?